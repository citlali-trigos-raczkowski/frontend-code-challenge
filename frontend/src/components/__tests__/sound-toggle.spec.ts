import { shallowMount, createLocalVue } from "@vue/test-utils";
import SoundToggle from "../sound-toggle.vue";
import VueRouter from "vue-router";
import { SoundToggleMockData } from "../../../tests/mocks/mock-props";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/heart-toggle.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(SoundToggle, {
      propsData: SoundToggleMockData,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sound source sets as expected", () => {
    const wrapper = shallowMount(SoundToggle, {
      propsData: SoundToggleMockData,
      localVue,
    });

    const aWrapper = wrapper.find("source");
    expect(aWrapper.attributes().src).toBe(SoundToggleMockData.sound);
  });

  it("sound plays on click", () => {
    const wrapper = shallowMount(SoundToggle, {
      propsData: SoundToggleMockData,
      localVue,
    });
    wrapper.find("v-btn-stub").trigger("click");
  });
});
