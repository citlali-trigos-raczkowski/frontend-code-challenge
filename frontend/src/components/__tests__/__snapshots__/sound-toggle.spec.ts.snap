// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/heart-toggle.vue renders correctly 1`] = `
<v-btn-stub
  activeclass=""
  alt="pokemon audio"
  color="inherit"
  icon="true"
  tag="button"
  type="button"
>
  <v-icon-stub>
    mdi-volume-high
  </v-icon-stub>
   
  <audio
    id="trigger-sound"
  >
    <source
      src="MockSoundSource"
      type="audio/mp3"
    />
    
    Your browser does not support the audio element.
  
  </audio>
</v-btn-stub>
`;
